sphinx/domains/python.py

```diff
@@ -984,7 +985,14 @@ def run(self) -> List[Node]:
         modname = self.arguments[0].strip()
         noindex = 'noindex' in self.options
         self.env.ref_context['py:module'] = modname
-        ret: List[Node] = []
+
+        content_node: Element = nodes.section()
+        with switch_source_input(self.state, self.content):
```

Comment:
Glad to see this fixed here too. I didn't look into it too deeply, but should `PyModule` just use the same mechanisms as the other Py* classes (subclassing from `PyObject`)?

Also I suppose the docs should be updated for this.

Reply:
Was this addressed (documenting the change to `py:module`)?

Reply:
Yes:

![image](https://user-images.githubusercontent.com/9087854/189745181-995a6635-29fb-4f62-9f00-cbc70424f891.png)

A